vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts

```diff
@@ -346,8 +346,19 @@ Please correct your PATH variable or make sure the 'open' utility is installed s
                 {
                     const registryQueryCommand = path.join(`${process.env.SystemRoot}`, `System32\\reg.exe`);
                     // /reg:32 is added because all keys on 64 bit machines are all put into the WOW node. They won't be on the WOW node on a 32 bit machine.
-                    const fullQuery = `${registryQueryCommand}`;
-                    const installRecordKeysOfXBit = await this.commandRunner.execute(CommandExecutor.makeCommand(registryQueryCommand, [`query`, `${query}`, `\/reg:32`]));
+                    const command = CommandExecutor.makeCommand(registryQueryCommand, [`query`, `${query}`, `\/reg:32`]);
+
+                    let installRecordKeysOfXBit = '';
+                    const oldReturnStatusSetting = this.commandRunner.returnStatus;
+                    this.commandRunner.returnStatus = true;
+                    const registryLookupStatusCode = await this.commandRunner.execute(command);
+                    this.commandRunner.returnStatus = oldReturnStatusSetting;
```

Comment:
Obviously this should be false, but I would prefer to keep this pattern so we can update them all at the same time if/when we want to fix this code.

Reply:
So if I understand correctly, what we're doing here is a two-phase lookup - in the first phase were invoking the command specifically to see if it returns successfully (since our current usage/API for executing commands here doesn't give us both exit code + stdin/stderr?). And then, once we are convinced that we can successfully get the reg query result, we invoke the query again in an execution mode where we get the result of the query back?

Reply:
That is 100% correct